open-company/open-company-web
# Remove sections preload

Card: https://trello.com/c/4cpAQlOK

NB: When leaving the expanded post view the posts are reloaded if the current container (AP/FU/section) has not been loaded yet or if the user left a comment in the post and is going back to the recent activity sort (which will change with the new comment). In all the other cases we don't reload the posts and also we reset the scroll to the previously saved position.

To test:
- load AP
- focus Network tab in dev tools
- switch to Section 1
- [x] do you see ghost screen? Good
- [x] do you see 2 requests for that section: one for the recent activity and one for the recently posted?
- [x] do you see the posts appear?
- switch to another Section 2
- now switch back to AP
- [x] no ghost screen
- [x] posts appear right away?
- [x] fast enough?
- now go back to Section 1
- [x] no ghost screen, posts appear right away, fast enough?
- now with another user add a post in Section 2
- with first user switch to Section 2
- [x] no ghost screen?
- [x] the new post is not there initially?
- [x] the new post appear?
- switch to AP
- [x] all posts are there?
- load a post page directly
- now click Back to AP
- [x] do you see AP posts?
- reload the post page
- now click on a section in the nav sidebar
- [x] do you see the posts?
- now open a post that should 404
- [x] do you see the dialog?
- dismiss the dialog
- [x] do you see the AP posts now?

Test for posts reload logic (as discussed on Monday's call):
- open a post url directly
- click back to button
- [x] do you see ghost screen?
- [x] posts appear when they are ready?
- open a post url directly
- click on a board in the sidebar (this could not be possible if #1049 has been merged)
- [x] do you see ghost screen?
- [x] posts appear when they are ready?
- open AP on recent activity sort
- scroll down to second page
- expand a post
- go back to AP
- [x] scroll position is preserved?
- [x] no reload happens? (check Network tab and look for url with `/entries?...`)
- open AP on recent activity sort
- scroll down to second page
- expand a post
- leave a comment
- click back to AP
- [x] back to the top?
- [x] posts are refreshed?
- open AP on recently posted
- scroll down to second page
- expand a post
- leave a comment
- click back to AP
- [x] scroll position is preserved?
- [x] posts are not reloaded?
